utils/wandb_logging/wandb_utils.py

```diff
@@ -243,16 +243,13 @@ def create_dataset_table(self, dataset, class_to_id, name='dataset'):
         table = wandb.Table(columns=["id", "train_image", "Classes", "name"])
         class_set = wandb.Classes([{'id': id, 'name': name} for id, name in class_to_id.items()])
         for si, (img, labels, paths, shapes) in enumerate(tqdm(dataset)):
-            height, width = shapes[0]
-            labels[:, 2:] = (xywh2xyxy(labels[:, 2:].view(-1, 4))) * torch.Tensor([width, height, width, height])
             box_data, img_classes = [], {}
             for cls, *xyxy in labels[:, 1:].tolist():
                 cls = int(cls)
-                box_data.append({"position": {"minX": xyxy[0], "minY": xyxy[1], "maxX": xyxy[2], "maxY": xyxy[3]},
+                box_data.append({"position": {"middle": [xyxy[0], xyxy[1]], "width": xyxy[2], "height": xyxy[3]},
                                  "class_id": cls,
                                  "box_caption": "%s" % (class_to_id[cls]),
-                                 "scores": {"acc": 1},
-                                 "domain": "pixel"})
+                                 "scores": {"acc": 1}})
                 img_classes[cls] = class_to_id[cls]
             boxes = {"ground_truth": {"box_data": box_data, "class_labels": class_to_id}}  # inference-space
             table.add_data(si, wandb.Image(paths, classes=class_set, boxes=boxes), json.dumps(img_classes),
```

Comment:
@glenn-jocher this the logging code. Works perfectly for datasets with all square images but boxes are misplaced in the raw dataset with unevenly sized images. The previous version also had the same problem

Reply:
@AyushExel are you sure this in the right space? There is inference space and image/native space, with coordinates available in both in test.py as `pred` and `predn`. Inference space is typically 640x640 while native space is the original image dimensions. If you are overlaying boxes on the original images then you want the coordinates in predn here:

Reply:
Native-space labels are here: https://github.com/ultralytics/yolov5/blob/4890499344e21950d985e1a77e84a0a4161d1db0/test.py#L185

Native-space predictions are here:
https://github.com/ultralytics/yolov5/blob/4890499344e21950d985e1a77e84a0a4161d1db0/test.py#L141-L142

Reply:
@glenn-jocher yes I'm looking for native space labels. I tried this based on your suggestion
```
        for si, (img, labels, paths, shapes) in enumerate(tqdm(dataset)):
            labels[:,2:] = xywh2xyxy(labels[:, 2:])
            scale_coords(img.shape[1:], labels[:,2:], shapes[0], shapes[1])  # native-space labels
            box_data, img_classes = [], {}
            for cls, *xyxy in labels[:, 1:].tolist():
                   #logging code
```
This also doesn't result in correct bboxes -> [output](https://wandb.ai/cayush/scaling_test/artifacts/dataset/train/8aff10b644180279066c/files/train.table.json)

Reply:
@glenn-jocher I was going through some of the `datasets.py` functions trying to get to the root of this scaling problem as the `dataset` object here is an instance of `LoadImagesAndLabels` class. Something that isn't clear to me is - Why do the original labels need scaling? Isn't these labels scanned from the `labels.txt` file, and if so, then these labels must be of correct dimension as they're read directly from the dataset source?

P.S. we'll be running a training class and a webinar for a few teams in industry in a week where we want to use this repo and integration heavily, so I wanted to get this fixed before that.

Reply:
@AyushExel the original labels (xywh normalized) are scaled to pixels in native space for the IoU operations during mAP computation. They are also separately scaled to grid units in inference space for loss computation.

Reply:
I guess the first question is what units and what reference space do you want the labels in, then I can point you to the right place.

Reply:
@glenn-jocher I think I need the images scaled to native space. Both pixel or percentage-of-image units are supported by wandb. These labels will just be used to make bboxes in W&B Tables like this  -> `Table(wandb.Image(paths[si], boxes=[# Native space labels data]))` .  My attempt at scaling correctly is mentioned in 2nd last comment above

Reply:
Hey perfect. You shouldn't need to scale or modify coordinates. The labels `tbox` predictions `predn` are already available in test.py in native pixels, you can access them after L185 and L142:

Native-space labels: https://github.com/ultralytics/yolov5/blob/4890499344e21950d985e1a77e84a0a4161d1db0/test.py#L185

Native-space predictions:
https://github.com/ultralytics/yolov5/blob/4890499344e21950d985e1a77e84a0a4161d1db0/test.py#L141-L142

Reply:
But I don't have access to `test.py` in `wandb_utils.py` which actually uploads the dataset. Here's what's happening in that file.
```
dataset = LoadImagesAndLabels(data['train'])
for si, (img, labels, paths, shapes) in enumerate(tqdm(dataset)):
    # Logging code here
```

Reply:
@AyushExel hmm maybe the best way to import the boxes is the same way that the ConfusionMatrix object is doing it here. All native-space predictions and labels for an image are passed in 1 line of code:
https://github.com/ultralytics/yolov5/blob/4200674a136a5589972f352790f76d3f37e98dd6/test.py#L186-L188

Reply:
okay. I'm trying this now. I'll let you know how it goes

Reply:
@glenn-jocher okay I tried everything I could think of and finally adding `rect=True, batch_size=1`  in `LoadImagesAndLabels` dataset worked 🙂  ! I saw that it was set to true in the dataloaders so I gave it a try and I've tested on multiple raw datasets and it's working fine now. I'll clean this PR up to include the changes tomorrow. 

Just wanted to ask what is it meant by rectangular training by setting `rect=True`?

Reply:
Oh great!! 

Rectangular dataloading is in place in PyTorch Hub models and the test.py dataloader for any batch size, and in the detect.py dataloaders for batch-size 1. This just means that the images in a batch are padded the minimum required to meet stride contraints and no more. You can see an example of this in test_batch0.jpg for example:
![image](https://user-images.githubusercontent.com/26833433/116588500-f126e000-a91b-11eb-9232-47fae18f4f97.png)

Reply:
@glenn-jocher I have made the changes. Can be merged now if everything looks good!